### ISSUE TITLE
installable_condition crashes client

### ISSUE THREAD

**Issue by mattydee99** [munki/munki#1201]
I made a mistake in the installable conditions buy specifying board-id (as specified in the release notes: https://github.com/munki/munki/releases/tag/v6.4.0) instead of board_id (as the var is defined in code) 

This crashed the client (error output attached).

Using the proper variable with the underscore remedied the issue, but I thought I should let you know. 
1 - update release notes (underscores not hyphens) 
2 - handle exception so that clients do not crash when improper vars are used in pkgsinfo files. 

```
ERROR: Unexpected error in updatecheck:
Traceback (most recent call last):
  File "/usr/local/munki/.managedsoftwareupdate.py", line 1271, in <module>
    main()
  File "/usr/local/munki/.managedsoftwareupdate.py", line 959, in main
    updatecheckresult = updatecheck.check(
  File "/usr/local/munki/munkilib/updatecheck/core.py", line 226, in check
    analyze.process_manifest_for_key(
  File "/usr/local/munki/munkilib/updatecheck/analyze.py", line 737, in process_manifest_for_key
    process_optional_install(item, cataloglist, installinfo)
  File "/usr/local/munki/munkilib/updatecheck/analyze.py", line 169, in process_optional_install
    item_pl = catalogs.get_item_detail(manifestitem, cataloglist,
  File "/usr/local/munki/munkilib/updatecheck/catalogs.py", line 596, in get_item_detail
    installable_condition_ok(item)):
  File "/usr/local/munki/munkilib/updatecheck/catalogs.py", line 539, in installable_condition_ok
    if not info.predicate_evaluates_as_true(
  File "/usr/local/munki/munkilib/info.py", line 935, in predicate_evaluates_as_true
    result = predicate.evaluateWithObject_(info_object)
ValueError: NSInvalidArgumentException - *** -[__NSPlaceholderArray initWithObjects:count:]: attempt to insert nil object from objects[0]
    Allowing idle sleep
```

**Comment by gregneagle** [comment]
I'm a bit confused here. On my machine, the "Conditions" key in /Library/Managed\ Installs/ManagedInstallReport.plist has "board-id" and "device-id" as keys; not "board_id" or "device_id".

**Comment by gregneagle** [comment]
Could you please share a pkginfo that triggers this issue? (and one that "fixes" it?)

**Comment by mattydee99** [comment]
> I'm a bit confused here. On my machine, the "Conditions" key in /Library/Managed\ Installs/ManagedInstallReport.plist has "board-id" and "device-id" as keys; not "board_id" or "device_id".

That is strange... maybe I missed something here. 

Here are the 2 strings in  <key>installable_condition</key>. The first string is problematic (crashes munki), and the second one seems to work. 

`<string>board-id IN {'Mac-0CFF9C7C2B63DF8D', 'Mac-112818653D3AABFC', 'Mac-1E7E29AD0135F9BC', 'Mac-226CB3C6A851A671', 'Mac-27AD2F918AE68F61', 'Mac-53FDB3D8DB8CA971', 'Mac-5F9802EFE386AA28', 'Mac-63001698E7A34814', 'Mac-7BA5B2D9E42DDD94', 'Mac-7BA5B2DFE22DDD8C', 'Mac-827FAC58A8FDFA22', 'Mac-827FB448E656EC26', 'Mac-937A206F2EE63C01', 'Mac-A61BADE1FDAD7B05', 'Mac-AA95B1DDAB278B95', 'Mac-AF89B6D9451A490B', 'Mac-CFF7D910A743CAAF', 'Mac-E1008331FDC96864', 'Mac-E7203C0F68AA0004'} OR device-id IN {'J132AP', 'J137AP', 'J140AAP', 'J140KAP', 'J152FAP', 'J160AP', 'J174AP', 'J185AP', 'J185FAP', 'J213AP', 'J214AP', 'J214KAP', 'J215AP', 'J223AP', 'J230AP', 'J230KAP', 'J680AP', 'J780AP', 'X589AMLUAP', 'X86LEGACYAP', 'J180dAP', 'J274AP', 'J293AP', 'J313AP', 'J314cAP', 'J314sAP', 'J316cAP', 'J316sAP', 'J375cAP', 'J375dAP', 'J413AP', 'J414cAP', 'J414sAP', 'J415AP', 'J416cAP', 'J416sAP', 'J433AP', 'J434AP', 'J456AP', 'J457AP', 'J473AP', 'J474sAP', 'J475cAP', 'J475dAP', 'J493AP', 'J504AP', 'J514cAP', 'J514mAP', 'J514sAP', 'J516cAP', 'J516mAP', 'J516sAP', 'VMA2MACOSAP', 'VMM-x86_64'}</string>`

`<string>board_id IN {'Mac-0CFF9C7C2B63DF8D', 'Mac-112818653D3AABFC', 'Mac-1E7E29AD0135F9BC', 'Mac-226CB3C6A851A671', 'Mac-27AD2F918AE68F61', 'Mac-53FDB3D8DB8CA971', 'Mac-5F9802EFE386AA28', 'Mac-63001698E7A34814', 'Mac-7BA5B2D9E42DDD94', 'Mac-7BA5B2DFE22DDD8C', 'Mac-827FAC58A8FDFA22', 'Mac-827FB448E656EC26', 'Mac-937A206F2EE63C01', 'Mac-A61BADE1FDAD7B05', 'Mac-AA95B1DDAB278B95', 'Mac-AF89B6D9451A490B', 'Mac-CFF7D910A743CAAF', 'Mac-E1008331FDC96864', 'Mac-E7203C0F68AA0004'} OR device_id IN {'J132AP', 'J137AP', 'J140AAP', 'J140KAP', 'J152FAP', 'J160AP', 'J174AP', 'J185AP', 'J185FAP', 'J213AP', 'J214AP', 'J214KAP', 'J215AP', 'J223AP', 'J230AP', 'J230KAP', 'J680AP', 'J780AP', 'X589AMLUAP', 'X86LEGACYAP', 'J180dAP', 'J274AP', 'J293AP', 'J313AP', 'J314cAP', 'J314sAP', 'J316cAP', 'J316sAP', 'J375cAP', 'J375dAP', 'J413AP', 'J414cAP', 'J414sAP', 'J415AP', 'J416cAP', 'J416sAP', 'J433AP', 'J434AP', 'J456AP', 'J457AP', 'J473AP', 'J474sAP', 'J475cAP', 'J475dAP', 'J493AP', 'J504AP', 'J514cAP', 'J514mAP', 'J514sAP', 'J516cAP', 'J516mAP', 'J516sAP', 'VMA2MACOSAP', 'VMM-x86_64'}</string>`

**Comment by gregneagle** [comment]
Looks like "identifiers"/keys with hyphens in their names trigger this problem. So the keys will have to be renamed to `device_id` and `board_id`. This, of course, means the new "board-id" and "device-id" facts are currently unusable. :-(

**Comment by mattydee99** [comment]
Thanks for the clarification! 
I'm not sure I understand exactly: Does this mean that using board_id simply quiets the error, and does not actually have the intended effect?

**Comment by gregneagle** [comment]
"Does this mean that using board_id simply quiets the error, and does not actually have the intended effect?" `board_id` doesn't exist/is not defined, so your predicate would return `false`.

**Comment by gregneagle** [comment]
Fixes are in the Munki6dev branch, and will be included in the Munki 6.4.2 release.

**Comment by gregneagle** [comment]
This bug is embarrassing since https://github.com/munki/munki/wiki/Conditional-Items#comparison-string-syntax clearly states:
> Attribute names must be "C style identifiers" -- that means they can contain only letters, digits, and underscores, and must start with a letter or underscore. "Group-A" is not a valid attribute name; "Group_A" is a valid attribute name.

**Comment by mattydee99** [comment]
> Fixes are in the Munki6dev branch, and will be included in the Munki 6.4.2 release.

Thank you!
